pkgs/development/embedded/pyocd/default.nix

```
@@ -0,0 +1,65 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, fetchpatch
+, capstone
+, cmsis-pack-manager
+, colorama
+, intelhex
+, intervaltree
+, natsort
+, prettytable
+, pyelftools
+, pylink-square
+, pyusb
+, pyyaml
+, typing-extensions
+, pytestCheckHook
+}:
+
+buildPythonPackage rec {
+  pname = "pyocd";
+  version = "0.34.1";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-Fpa2IEsLOQ8ylGI/5D6h+22j1pvrvE9IMIyhCtyM6qU=";
+  };
+
+  patches = [
+    # Remove requirement for libusb-package (https://github.com/pyocd/pyOCD/pull/1332)
+    (fetchpatch {
+      url = "https://github.com/pyocd/pyOCD/commit/0b980cf253e3714dd2eaf0bddeb7172d14089649.patch";
+      sha256 = "sha256-B2+50VntcQELeakJbCeJdgI1iBU+h2NkXqba+LRYa/0=";
+    })
+  ];
+
+  propagatedBuildInputs = [
+    capstone
+    cmsis-pack-manager
+    colorama
+    intelhex
+    intervaltree
+    natsort
+    prettytable
+    pyelftools
+    pylink-square
+    pyusb
+    pyyaml
+    typing-extensions
+  ];
```

```suggestion
  ] ++ lib.optionals (!stdenv.isLinux) [
    hidapi
  ];
```

...and obviously also add it to the args at the top.